### PR TITLE
oep(csi): add controller patterns to csi volume provisioning proposal

### PR DESCRIPTION
**Pull request by AmitKumarDas**
This commit adds controller patterns section to csi volume provisioning proposal. These patterns are a derivation from existing kubernetes controller projects.

Signed-off-by: AmitKumarDas <amit.das@mayadata.io>

### CODE REVIEW

File: contribute/design/1.x/csi/20190606-csi-volume-provisioning.md [hunk 1]
```diff
@@ -457,6 +458,56 @@ spec:
...
+    - True
+    - False
+    - Unknown
+  - An item within a condition can be added, updated & deleted by the resource’s own controller
+  - An item within a condition can be updated by a separate controller
+    - The resource’s own controller still holds the right to delete this condition item
+
+Below is a sample schema snippet for `CStorVolumeClaim` resource
+```go
+type CStorVolumeClaim struct {
```
Comment (prateekpandey14) [inline contribute/design/1.x/csi/20190606-csi-volume-provisioning.md]: nit : `CStorVolumeClaimStatus` ?

Comment (AmitKumarDas) [reply contribute/design/1.x/csi/20190606-csi-volume-provisioning.md]: Done

File: contribute/design/1.x/csi/20190606-csi-volume-provisioning.md [hunk 2]
```diff
@@ -457,6 +458,56 @@ spec:
...
+  - Bound
+- Status.Phase is only set during creation
+  - Once the phase is Bound it should never revert to Pending
+- Status.Conditions will be used to track an on-going operation or sub status-es
+  - It will be an array
+  - It will have ConditionStatus as a field which can have below values:
+    - True
+    - False
+    - Unknown
+  - An item within a condition can be added, updated & deleted by the resource’s own controller
```
Comment (vishnuitta) [inline contribute/design/1.x/csi/20190606-csi-volume-provisioning.md]: in this case of CVC, its CSINode?

Comment (AmitKumarDas) [reply contribute/design/1.x/csi/20190606-csi-volume-provisioning.md]: It can be any CSINode. In other words the one which that receives the request.

**Pull request merged**